libbeat/common/mapstr_test.go

```diff
@@ -43,6 +43,126 @@ func TestMapStrUnion(t *testing.T) {
 	assert.Equal(c, MapStr{"a": 1, "b": 3, "c": 4})
 }
 
+func TestMapStrCopyTo(t *testing.T) {
+	assert := assert.New(t)
+
+	m := MapStr{
+		"a": MapStr{
+			"a1": 2,
+			"a2": 3,
+		},
+		"b": 2,
+		"c": MapStr{
+			"c1": 1,
+			"c2": 2,
+			"c3": MapStr{
+				"c31": 1,
+				"c32": 2,
+			},
+		},
+	}
+	c := MapStr{}
+
+	err := m.CopyTo(c, "dd")
+	assert.Equal(nil, err)
+	assert.Equal(MapStr{}, c)
+
+	err = m.CopyTo(c, "a")
+	assert.Equal(nil, err)
+	assert.Equal(MapStr{"a": MapStr{"a1": 2, "a2": 3}}, c)
+
+	err = m.CopyTo(c, "c.c1")
+	assert.Equal(nil, err)
+	assert.Equal(MapStr{"a": MapStr{"a1": 2, "a2": 3}, "c": MapStr{"c1": 1}}, c)
+
+	err = m.CopyTo(c, "b")
+	assert.Equal(nil, err)
+	assert.Equal(MapStr{"a": MapStr{"a1": 2, "a2": 3}, "c": MapStr{"c1": 1}, "b": 2}, c)
+
+	err = m.CopyTo(c, "c.c3.c32")
+	assert.Equal(nil, err)
+	assert.Equal(MapStr{"a": MapStr{"a1": 2, "a2": 3}, "c": MapStr{"c1": 1, "c3": MapStr{"c32": 2}}, "b": 2}, c)
+}
+
+func TestMapStrDelete(t *testing.T) {
+	assert := assert.New(t)
+
+	m := MapStr{
+		"c": MapStr{
+			"c1": 1,
+			"c2": 2,
+			"c3": MapStr{
+				"c31": 1,
+				"c32": 2,
+			},
+		},
+	}
+
+	err := m.Delete("c.c2")
+	assert.Equal(nil, err)
+	assert.Equal(MapStr{"c": MapStr{"c1": 1, "c3": MapStr{"c31": 1, "c32": 2}}}, m)
+
+	err = m.Delete("c.c2.c21")
```

Comment:
This removes an non existent field. So it is expected that no error is returned here?